packages/frontend-api/src/Model/Resolver/Products/Search/ProductSearchResultsProvider.php

```diff
@@ -51,7 +51,16 @@ function ($offset, $limit) use ($search, $productFilterData) {
             $this->productFacade->getFilteredProductsCountOnCurrentDomain($productFilterData, $search),
             $argument,
             $productFilterData,
-            ProductsQuery::getOrderingModeFromArgument($argument),
+            ProductsQuery::getDefaultOrderingMode($argument),
```

Comment:
Fixed

Reply:
please consider extensibility. With the static method it is not possible to change the behavior in the project

Reply:
I have moved it into the trait

Reply:
hmm, but it is not possible to extend a trait, right? What if someone wants to change e.g. default ordering mode now? How can he overwrite `getDefaultOrderingMode`method?

Reply:
I believe it could be reverted into the previous state and just wrap the static call into a protected method in the classes where you use `ProductsQuery::getOrderingModeFromArgument($argument),`?

So you end up with `$this->getOrderingMode($argument)` and
```
protected function getOrderingMode($argument)
{
    return ProductsQuery::getOrderingModeFromArgument($argument);
}
```

and maybe this protected method can be static as well. Or maybe I misunderstood what @grossmannmartin meant?

Reply:
ah, my bad, this does not make sense. My original thought was creating new `ProductOrderingModeProvider` and use it via autowiring without static methods

Reply:
When ProductsQuery is extended, the `ProductsQuery::getOrderingModeFromArgument($argument)` will always call the parent method in the shopsys/framework package context.

Reply:
I have introduced new service for this